app/src/main/java/ca/dal/cs/athletemonitor/athletemonitor/AccountManager.java

```diff
@@ -0,0 +1,226 @@
+        //otherwise, delete any existing entry
+        if (online) {
+            usersReference.child(username).setValue(true);
+        } else {
+            usersReference.child(username).removeValue();
+        }
+    }
+
+    /**
+     * Determines whether or not the specified user exists in the database
+     *
+     * @param username Name of the user to look up
+     *
+     */
+    public static void userExists(final String username, final UserExistsListener listener) {
+        //retrieve a reference to the users node
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference usersReference = database.getReference("users/");
+
+        //attach a listener for data changes of the users reference.  this will occur when
+        //the reference is populated
+        usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                //if the username is a child of the users node then fire the user exists event
+                //indicating true, otherwise, fire event with false outcome
+                if (dataSnapshot.hasChild(username)) {
+                    if (listener != null) listener.onUserExists(true);
+                }
+                else {
+                    if (listener != null) listener.onUserExists(false);
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+    }
+
+    /**
+     * Creates a new user account
+     *
+     * @param newUser User details
+     */
+    public static void createUser(final User newUser) {
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference usersReference = database.getReference("users");
+        usersReference.child(newUser.getUsername()).setValue(newUser, null);
+    }
+
+    /**
+     * Creates a new user account and called the specific listener when complete.
+     *
+     * Note: This method causes a destructive write.  Callers should ensure that the user does
+     * not exist prior to calling this method.
+     *
+     * @param newUser User details
+     * @param userObjectListener Callback for completion
+     */
+    public static void createUser(final User newUser, final UserObjectListener userObjectListener) {
+        //get a reference to the users node
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        DatabaseReference usersReference = database.getReference("users");
+
+        //attempt to write the data
+        usersReference.child(newUser.getUsername()).setValue(newUser, new DatabaseReference.CompletionListener() {
+            @Override
+            public void onComplete(DatabaseError databaseError, DatabaseReference databaseReference) {
+                //if we have a listener, report the result
+                if (userObjectListener != null) {
```

Comment:
Similar to above (with the difference being that this is a DatabaseReference.CompletionListener, so perhaps the answer is different here), would you expect this to be null under normal conditions?

Reply:
In this case, the null check is on the createUser listener.  A null check is not necessary on the CompletionListener since it is being created with no possibility of being null.